src/common/perf_counters_cache.cc

```diff
@@ -0,0 +1,116 @@
+#include "common/perf_counters_cache.h"
+#include "common/perf_counters_key.h"
+
+namespace ceph::perf_counters {
+
+void PerfCountersCache::check_key(const std::string &key) {
+  std::string_view key_name = ceph::perf_counters::key_name(key);
+  // return false for empty key name
+  assert(key_name != "");
```

Comment:
please use `ceph_assert()`s instead of raw `assert()`s

Reply:
> I'm concerned about introducing ceph_assert(...), which is not removed from release builds, for things like this. the original C assert feature doesn't have this problem

i agree with @mattbenjamin's comment above, i think this is fine as a normal `assert()`

it's probably time to revisit Sage's idea of making `ceph_assert()` conditional on `NDEBUG` - that's why he added a `ceph_assert_always()` in https://github.com/ceph/ceph/pull/9969 (in 2016)